lib/utils/cluster-utils.ts

```diff
@@ -0,0 +1,51 @@
+import { Construct } from "@aws-cdk/core";
+import * as customResource from '@aws-cdk/custom-resources';
+
+interface Tag {
+  Key: string;
+  Value: string;
+}
+
+/**
+ * Creates the node termination tag for the ASG
+ * @param scope
+ * @param autoScalingGroup 
+ */
+ export function tagAsg(scope: Construct, autoScalingGroup: string, tags: Tag[]): void {
+  let tagList: {
+    Key: string;
+    Value: string;
+    PropagateAtLaunch: boolean;
+    ResourceId: string;
+    ResourceType: string;
+  }[] = [];
+
+  tags.forEach((tag) => {
+    tagList.push({
+      Key: tag.Key,
+      Value: tag.Value,
+      PropagateAtLaunch : true,
+      ResourceId: autoScalingGroup,
+      ResourceType: 'auto-scaling-group'
+    });
+  });
+
+  const callProps: customResource.AwsSdkCall = {
+    service: 'AutoScaling',
+    action: 'createOrUpdateTags',
+    parameters: {
+      Tags: tagList
+    },
+    physicalResourceId: customResource.PhysicalResourceId.of(
+      `${autoScalingGroup}-asg-tag`
+    )
+  };
+
+  new customResource.AwsCustomResource(scope, 'asg-tag', {
```

Comment:
No native CDK way to tag ASG other then custom resource? is there an issue on it?

Reply:
There doesn't seem to be a way to tag resources after they have been created. I saw that @youngjeong46 added a util to tag subnets. We could create a separate construct for the `custom-resource` so it only provisions one resource. I fast follow that as a separate issue.

Reply:
https://github.com/aws-quickstart/ssp-amazon-eks/issues/272